plenario_client/responses.py

```diff
@@ -46,6 +47,58 @@ def __repr__(self) -> str:
         return '<Response {url}>'.format(url=self.current_url)
 
 
+class TimeRange:
+    """
+    ``TimeRange`` is a wrapper around the time_range response from the API. It encapsulates
+    the lower and upper bounds into datetime objects, and the lower and upper inclusiveness
+    into bools. Note that on initialization, all ranges are normalized to lower inclusive and
+    upper exclusive
+    """
+
+    def __init__(self, time_range_json: dict):
+        """
+        Initializes a new ``TimeRange``
+        :param time_range_json: JSON dictionary containing upper and lower bounds,
+        and their inclusiveness.
+        """
+        self.lower_bound: datetime = parse_datetime(time_range_json.get('lower'))
+        self.upper_bound: datetime = parse_datetime(time_range_json.get('upper'))
+        self.lower_inclusive: bool = time_range_json.get('lower_inclusive')
+        self.upper_inclusive: bool = time_range_json.get('upper_inclusive')
+
+        if not self.lower_inclusive:
+            self.lower_bound = self.lower_bound + timedelta(seconds=1)
+            self.lower_inclusive = True
+
+        if self.upper_inclusive:
+            self.upper_bound = self.upper_bound + timedelta(seconds=1)
+            self.upper_inclusive = False
+
+    def __lt__(self, other):
```

Comment:
I'm going to write one comment for this, but it applies to all the comparison methods. Your logic here is a little off. Here's a visual that might help:

```
0 --------|================ Range 1 ================|------------------------- 100
0 ------------------|================ Range 2 ================|--------------- 100
```

Would you say that the first range is strictly less than the second? I wouldn't -- it's upper bound is greater than the lower bound of the other.